tests/functional/test_receipt.py

```diff
@@ -115,13 +115,16 @@ def test_decode_logs_unspecified_abi_gets_all_logs(owner, contract_instance):
 
 def test_get_failed_receipt(owner, vyper_contract_instance, eth_tester_provider):
     # Setting to '5' always fails.
-    transaction = vyper_contract_instance.setNumber.as_transaction(
-        5, sender=owner, gas_limit=100000
-    )
-    receipt = owner.call(transaction)
+    receipt = vyper_contract_instance.setNumber(5, sender=owner, gas_limit=100000)
     assert receipt.failed
 
 
+def test_get_failed_receipt_raise(owner, vyper_contract_instance, eth_tester_provider):
+    # Setting to '5' always fails.
+    with pytest.raises(TransactionError):
+        vyper_contract_instance.setNumber(5, sender=owner, gas_limit=100000, raise_on_fail=True)
```

Comment:
Where is the receipt here?

Reply:
@unparalleled-js this raises, so it doesn't return

Reply:
Ah ok I think I was confused by the test name, `test_get_failed_receipt_raise` .

Can we call it something like `test_contract_method_raise_on_fail()`

Reply:
Updated in f42fa62a